### DISCUSSION TITLE
Snapshots take exceedingly long

### DISCUSSION THREAD

**Discussion by marcoow** [percy/percy-storybook#703] [Support] [answered]
## The problem

We're running into a scenario where snapshots take exceedingly long to generate. The first few are generate in 7ms or so, then it takes 500ms, then 3000ms, 12000ms, and finally 200000ms. We're snapshotting a number of super simple React components in Storybook and I cannot see anything that could cause this and am a bit lost.

Clearly something is leaking in some way and the cost of that is adding up exponentially somehow. I don't know whether this is a known bug? I'm attaching the Storybook build we're using as well as the Percy log (I canceled at some point)

## Environment

- Node version: 16.19.0
- `@percy/cli` version: 1.17.0
- Version of Percy SDK you’re using: @percy/storybook@4.3.4
- If needed, a build or snapshot ID: https://percy.io/simplabs-com/Gravity/builds/24383036
- OS version: macOS 13.1 (times out on GitHub Actions as well – whatever Ubuntu version that is)
- Type of shell command-line [interface]: zsh/GtHub Actions

## Details

–

## Debug logs

<details>
<summary></summary>
yarn percy --debug
yarn run v1.22.19
$ yarn workspace @gravityci/dashboard percy --debug
$ percy storybook ./storybook-static --debug

[percy] A new version of @percy/cli is available! 1.17.0 -> 1.19.1-alpha.0

[percy:config] Found config file: .percy.js (883ms)
[percy:config] Invalid config: (32ms)
[percy:config] - snapshot.widths: must be an array, received a number (0ms)
[percy:config] Using config:
{
  version: 2,
  snapshot: {
    minHeight: 1024
  }
} (2ms)
[percy:storybook] Requesting Storybook: http://localhost:59641 (43ms)
[percy:core:browser] Launching browser (25ms)
[percy:core:browser] Browser connected [61511]: HeadlessChrome/96.0.4664.0 (290ms)
[percy:core] Percy has started! (1ms)
[percy:core:page] Page created (7ms)
[percy:core:page] Page created (6ms)
[percy:core:page] Navigate to: http://localhost:59641/?path=/settings/about (243ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html (3ms)
[percy:core:page] Page navigated (43ms)
[percy:core:page] Page navigated (440ms)
[percy:core:page] Page closed (16ms)
[percy:core:page] Page closed (5000ms)
[percy:core:page] Page created (42ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html (97ms)
[percy:core:page] Page navigated (328ms)
[percy:core:page] Taking snapshot: Gravity Dashboard/Button: Approval (60ms)
[percy:core:discovery] Wait for 100ms idle (1ms)
[percy:core:page] Inject @percy/dom (105ms)
[percy:core:page] Serialize DOM (4ms)
[percy:core:snapshot] --------- (30ms)
[percy:core:snapshot] Received snapshot: Gravity Dashboard/Button: Approval (0ms)
[percy:core:snapshot] - url: http://localhost:59641/iframe.html?id=gravity-dashboard-button--approval (0ms)
[percy:core:snapshot] - widths: 375px, 1280px (1ms)
[percy:core:snapshot] - minHeight: 1024px (0ms)
[percy:core:snapshot] - discovery.allowedHostnames: localhost (0ms)
[percy:core:snapshot] - domSnapshot: true (0ms)
[percy:core] Discovering resources: Gravity Dashboard/Button: Approval (1ms)
[percy:core:page] Page created (4ms)
[percy:core:page] Taking snapshot: Gravity Dashboard/Button: Approval Hover (32ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Resize page to 375x1024 @1x (69ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html?id=gravity-dashboard-button--approval (7ms)
[percy:core:discovery] Handling request: http://localhost:59641/iframe.html?id=gravity-dashboard-button--approval (3ms)
[percy:core:discovery] - Serving root resource (0ms)
[percy:core:discovery] Handling request: http://localhost:59641/fonts/circularstd-book.woff2 (11ms)
[percy:core:page] Serialize DOM (13ms)
[percy:core:snapshot] --------- (7ms)
[percy:core:snapshot] Received snapshot: Gravity Dashboard/Button: Approval Hover (0ms)
[percy:core:snapshot] - url: http://localhost:59641/iframe.html?id=gravity-dashboard-button--approval-hover (0ms)
[percy:core:snapshot] - widths: 375px, 1280px (0ms)
[percy:core:snapshot] - minHeight: 1024px (0ms)
[percy:core:snapshot] - discovery.allowedHostnames: localhost (0ms)
[percy:core:snapshot] - domSnapshot: true (0ms)
[percy:core] Discovering resources: Gravity Dashboard/Button: Approval Hover (1ms)
[percy:core:page] Page created (4ms)
[percy:core:discovery] Processing resource: http://localhost:59641/fonts/circularstd-book.woff2 (29ms)
[percy:core:page] Taking snapshot: Gravity Dashboard/Button: Approval Active (1ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Page navigated (1ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:discovery] - Requesting asset directly (3ms)
[percy:core:discovery] - sha: c275d17a94d65cf98ecdbf84ba94f03304ad54128c06777dfb7e960e44cba509 (12ms)
[percy:core:discovery] - mimetype: font/woff2 (0ms)
[percy:core:page] Resize page to 375x1024 @1x (68ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html?id=gravity-dashboard-button--approval-hover (7ms)
[percy:core:discovery] Handling request: http://localhost:59641/iframe.html?id=gravity-dashboard-button--approval-hover (3ms)
[percy:core:discovery] - Serving root resource (0ms)
[percy:core:page] Serialize DOM (7ms)
[percy:core:snapshot] --------- (7ms)
[percy:core:snapshot] Received snapshot: Gravity Dashboard/Button: Approval Active (0ms)
[percy:core:snapshot] - url: http://localhost:59641/iframe.html?id=gravity-dashboard-button--approval-active (0ms)
[percy:core:snapshot] - widths: 375px, 1280px (0ms)
[percy:core:snapshot] - minHeight: 1024px (0ms)
[percy:core:snapshot] - discovery.allowedHostnames: localhost (0ms)
[percy:core:snapshot] - domSnapshot: true (0ms)
[percy:core] Discovering resources: Gravity Dashboard/Button: Approval Active (1ms)
[percy:core:discovery] Handling request: http://localhost:59641/fonts/circularstd-book.woff2 (2ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:page] Page created (2ms)
[percy:core:page] Resize page to 1280x1024 @1x (9ms)
[percy:core:discovery] Wait for 100ms idle (7ms)
[percy:core:page] Page navigated (24ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Taking snapshot: Gravity Dashboard/Button: Approval Disabled (4ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Resize page to 375x1024 @1x (69ms)
[percy:core:page] Page closed (6ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html?id=gravity-dashboard-button--approval-active (3ms)
[percy:core:discovery] Handling request: http://localhost:59641/iframe.html?id=gravity-dashboard-button--approval-active (3ms)
[percy:core:discovery] - Serving root resource (0ms)
[percy:core:page] Resize page to 1280x1024 @1x (15ms)
[percy:core:discovery] Wait for 100ms idle (4ms)
[percy:core:page] Serialize DOM (3ms)
[percy:core:snapshot] --------- (8ms)
[percy:core:snapshot] Received snapshot: Gravity Dashboard/Button: Approval Disabled (0ms)
[percy:core:snapshot] - url: http://localhost:59641/iframe.html?id=gravity-dashboard-button--approval-disabled (1ms)
[percy:core:snapshot] - widths: 375px, 1280px (0ms)
[percy:core:snapshot] - minHeight: 1024px (0ms)
[percy:core:snapshot] - discovery.allowedHostnames: localhost (0ms)
[percy:core:snapshot] - domSnapshot: true (0ms)
[percy:core] Discovering resources: Gravity Dashboard/Button: Approval Disabled (1ms)
[percy:core:discovery] Handling request: http://localhost:59641/fonts/circularstd-book.woff2 (3ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:page] Page created (2ms)
[percy:core:page] Taking snapshot: Gravity Dashboard/Button: Login (14ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Page navigated (22ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Page closed (48ms)
[percy:core:page] Resize page to 375x1024 @1x (17ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html?id=gravity-dashboard-button--approval-disabled (6ms)
[percy:core:discovery] Handling request: http://localhost:59641/iframe.html?id=gravity-dashboard-button--approval-disabled (3ms)
[percy:core:discovery] - Serving root resource (0ms)
[percy:core:page] Serialize DOM (5ms)
[percy:core:snapshot] --------- (7ms)
[percy:core:snapshot] Received snapshot: Gravity Dashboard/Button: Login (0ms)
[percy:core:snapshot] - url: http://localhost:59641/iframe.html?id=gravity-dashboard-button--login (0ms)
[percy:core:snapshot] - widths: 375px, 1280px (0ms)
[percy:core:snapshot] - minHeight: 1024px (0ms)
[percy:core:snapshot] - discovery.allowedHostnames: localhost (0ms)
[percy:core:snapshot] - domSnapshot: true (0ms)
[percy:core] Discovering resources: Gravity Dashboard/Button: Login (1ms)
[percy:core:discovery] Handling request: http://localhost:59641/fonts/circularstd-book.woff2 (0ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:page] Page created (4ms)
[percy:core:page] Resize page to 1280x1024 @1x (9ms)
[percy:core:discovery] Wait for 100ms idle (2ms)
[percy:core:page] Taking snapshot: Build Details/FileTree/File: Primary (7ms)
[percy:core:discovery] Wait for 100ms idle (1ms)
[percy:core:page] Page navigated (31ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Resize page to 375x1024 @1x (56ms)
[percy:core:page] Page closed (7ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html?id=gravity-dashboard-button--login (1ms)
[percy:core:discovery] Handling request: http://localhost:59641/iframe.html?id=gravity-dashboard-button--login (3ms)
[percy:core:discovery] - Serving root resource (0ms)
[percy:core:page] Serialize DOM (3ms)
[percy:core:snapshot] --------- (6ms)
[percy:core:snapshot] Received snapshot: Build Details/FileTree/File: Primary (0ms)
[percy:core:snapshot] - url: http://localhost:59641/iframe.html?id=build-details-filetree-file--primary (0ms)
[percy:core:snapshot] - widths: 375px, 1280px (0ms)
[percy:core:snapshot] - minHeight: 1024px (0ms)
[percy:core:snapshot] - discovery.allowedHostnames: localhost (0ms)
[percy:core:snapshot] - domSnapshot: true (0ms)
[percy:core] Discovering resources: Build Details/FileTree/File: Primary (1ms)
[percy:core:discovery] Handling request: http://localhost:59641/icons/github.svg (1ms)
[percy:core:page] Page created (3ms)
[percy:core:discovery] Handling request: http://localhost:59641/fonts/circularstd-book.woff2 (17ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:page] Resize page to 1280x1024 @1x (3ms)
[percy:core:discovery] Processing resource: http://localhost:59641/icons/github.svg (1ms)
[percy:core:page] Taking snapshot: Build Details/FileTree/File: Growing (2ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:discovery] Wait for 100ms idle (1ms)
[percy:core:discovery] - sha: fc1c7d0b18ef082e10a2f983fb4a5c208dd37193387765ca3f0a00d84b401856 (28ms)
[percy:core:discovery] - mimetype: image/svg+xml (0ms)
[percy:core:page] Page navigated (1ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Resize page to 375x1024 @1x (52ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html?id=build-details-filetree-file--primary (7ms)
[percy:core:discovery] Handling request: http://localhost:59641/iframe.html?id=build-details-filetree-file--primary (2ms)
[percy:core:discovery] - Serving root resource (0ms)
[percy:core:discovery] Handling request: http://localhost:59641/static/media/file.a0c51867.svg (8ms)
[percy:core:page] Serialize DOM (3ms)
[percy:core:page] Page closed (2ms)
[percy:core:discovery] Handling request: http://localhost:59641/fonts/circularstd-book.woff2 (2ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:discovery] Processing resource: http://localhost:59641/static/media/file.a0c51867.svg (3ms)
[percy:core:snapshot] --------- (5ms)
[percy:core:snapshot] Received snapshot: Build Details/FileTree/File: Growing (0ms)
[percy:core:snapshot] - url: http://localhost:59641/iframe.html?id=build-details-filetree-file--growing (0ms)
[percy:core:snapshot] - widths: 375px, 1280px (0ms)
[percy:core:snapshot] - minHeight: 1024px (0ms)
[percy:core:snapshot] - discovery.allowedHostnames: localhost (0ms)
[percy:core:snapshot] - domSnapshot: true (0ms)
[percy:core] Discovering resources: Build Details/FileTree/File: Growing (1ms)
[percy:core:page] Page created (4ms)
[percy:core:page] Resize page to 1280x1024 @1x (11ms)
[percy:core:discovery] Wait for 100ms idle (7ms)
[percy:core:page] Taking snapshot: Build Details/FileTree/File: Added (5ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:discovery] - sha: 07d284089a19c1c84956704be4bcd1d546298939f5df520c9e74fa24a16ccce4 (13ms)
[percy:core:discovery] - mimetype: image/svg+xml (0ms)
[percy:core:page] Page navigated (8ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Resize page to 375x1024 @1x (57ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html?id=build-details-filetree-file--growing (7ms)
[percy:core:discovery] Handling request: http://localhost:59641/iframe.html?id=build-details-filetree-file--growing (3ms)
[percy:core:discovery] - Serving root resource (0ms)
[percy:core:page] Page closed (10ms)
[percy:core:discovery] Handling request: http://localhost:59641/static/media/file.a0c51867.svg (1ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:page] Serialize DOM (2ms)
[percy:core:discovery] Handling request: http://localhost:59641/fonts/circularstd-book.woff2 (6ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:snapshot] --------- (15ms)
[percy:core:snapshot] Received snapshot: Build Details/FileTree/File: Added (0ms)
[percy:core:snapshot] - url: http://localhost:59641/iframe.html?id=build-details-filetree-file--added (0ms)
[percy:core:snapshot] - widths: 375px, 1280px (0ms)
[percy:core:snapshot] - minHeight: 1024px (0ms)
[percy:core:snapshot] - discovery.allowedHostnames: localhost (0ms)
[percy:core:snapshot] - domSnapshot: true (0ms)
[percy:core] Discovering resources: Build Details/FileTree/File: Added (2ms)
[percy:core:page] Resize page to 1280x1024 @1x (0ms)
[percy:core:page] Page created (5ms)
[percy:core:discovery] Wait for 100ms idle (1ms)
[percy:core:page] Page navigated (16ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Taking snapshot: Build Details/FileTree/File: Deleted (14ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Page closed (73ms)
[percy:core:page] Resize page to 375x1024 @1x (2ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html?id=build-details-filetree-file--added (6ms)
[percy:core:discovery] Handling request: http://localhost:59641/iframe.html?id=build-details-filetree-file--added (3ms)
[percy:core:discovery] - Serving root resource (0ms)
[percy:core:page] Resize page to 1280x1024 @1x (2ms)
[percy:core:discovery] Wait for 100ms idle (3ms)
[percy:core:discovery] Handling request: http://localhost:59641/static/media/file.a0c51867.svg (7ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:page] Serialize DOM (4ms)
[percy:core:discovery] Handling request: http://localhost:59641/fonts/circularstd-book.woff2 (23ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:snapshot] --------- (39ms)
[percy:core:snapshot] Received snapshot: Build Details/FileTree/File: Deleted (0ms)
[percy:core:snapshot] - url: http://localhost:59641/iframe.html?id=build-details-filetree-file--deleted (0ms)
[percy:core:snapshot] - widths: 375px, 1280px (0ms)
[percy:core:snapshot] - minHeight: 1024px (0ms)
[percy:core:snapshot] - discovery.allowedHostnames: localhost (0ms)
[percy:core:snapshot] - domSnapshot: true (0ms)
[percy:core] Discovering resources: Build Details/FileTree/File: Deleted (2ms)
[percy:core:page] Page navigated (0ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Page created (4ms)
[percy:core:page] Page closed (24ms)
[percy:core:page] Taking snapshot: Build Details/FileTree/File: Reducing (10ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Resize page to 1280x1024 @1x (62ms)
[percy:core:discovery] Wait for 100ms idle (3ms)
[percy:core:page] Resize page to 375x1024 @1x (7ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html?id=build-details-filetree-file--deleted (7ms)
[percy:core:discovery] Handling request: http://localhost:59641/iframe.html?id=build-details-filetree-file--deleted (2ms)
[percy:core:discovery] - Serving root resource (0ms)
[percy:core:discovery] Handling request: http://localhost:59641/static/media/file.a0c51867.svg (13ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:page] Serialize DOM (7ms)
[percy:core:discovery] Handling request: http://localhost:59641/fonts/circularstd-book.woff2 (3ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:page] Page navigated (55ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Page closed (8ms)
[percy:core:page] Resize page to 1280x1024 @1x (92ms)
[percy:core:discovery] Wait for 100ms idle (2ms)
[percy:core:snapshot] --------- (64ms)
[percy:core:snapshot] Received snapshot: Build Details/FileTree/File: Reducing (0ms)
[percy:core:snapshot] - url: http://localhost:59641/iframe.html?id=build-details-filetree-file--reducing (0ms)
[percy:core:snapshot] - widths: 375px, 1280px (0ms)
[percy:core:snapshot] - minHeight: 1024px (0ms)
[percy:core:snapshot] - discovery.allowedHostnames: localhost (0ms)
[percy:core:snapshot] - domSnapshot: true (0ms)
[percy:core] Discovering resources: Build Details/FileTree/File: Reducing (2ms)
[percy:core:page] Page created (4ms)
[percy:core:page] Page closed (34ms)
[percy:core:page] Taking snapshot: Build Details/FileTree/Tree: Primary (2ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Resize page to 375x1024 @1x (63ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html?id=build-details-filetree-file--reducing (7ms)
[percy:core:discovery] Handling request: http://localhost:59641/iframe.html?id=build-details-filetree-file--reducing (2ms)
[percy:core:discovery] - Serving root resource (0ms)
[percy:core:discovery] Handling request: http://localhost:59641/static/media/file.a0c51867.svg (15ms)
[percy:core:discovery] - Resource cache hit (1ms)
[percy:core:page] Serialize DOM (13ms)
[percy:core:discovery] Handling request: http://localhost:59641/fonts/circularstd-book.woff2 (25ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:page] Page navigated (34ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Resize page to 1280x1024 @1x (100ms)
[percy:core:discovery] Wait for 100ms idle (1ms)
[percy:core:page] Page closed (103ms)
[percy:core:snapshot] --------- (530ms)
[percy:core:snapshot] Received snapshot: Build Details/FileTree/Tree: Primary (0ms)
[percy:core:snapshot] - url: http://localhost:59641/iframe.html?id=build-details-filetree-tree--primary (0ms)
[percy:core:snapshot] - widths: 375px, 1280px (0ms)
[percy:core:snapshot] - minHeight: 1024px (0ms)
[percy:core:snapshot] - discovery.allowedHostnames: localhost (0ms)
[percy:core:snapshot] - domSnapshot: true (0ms)
[percy:core] Discovering resources: Build Details/FileTree/Tree: Primary (3ms)
[percy:core:page] Page created (3ms)
[percy:core:page] Taking snapshot: Build Details/FileTree/Tree: With Multiple Levels (50ms)
[percy:core:discovery] Wait for 100ms idle (1ms)
[percy:core:page] Resize page to 375x1024 @1x (56ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html?id=build-details-filetree-tree--primary (5ms)
[percy:core:discovery] Handling request: http://localhost:59641/iframe.html?id=build-details-filetree-tree--primary (3ms)
[percy:core:discovery] - Serving root resource (0ms)
[percy:core:discovery] Handling request: http://localhost:59641/static/media/folder.ac056cd0.svg (22ms)
[percy:core:discovery] Handling request: http://localhost:59641/static/media/file.a0c51867.svg (2ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:page] Serialize DOM (14ms)
[percy:core:discovery] Handling request: http://localhost:59641/fonts/circularstd-book.woff2 (18ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:discovery] Processing resource: http://localhost:59641/static/media/folder.ac056cd0.svg (3ms)
[percy:core:discovery] - sha: f646d9b35401638d52b78565ee66614e6836029fa2320ee8ecbcf3b8136fd789 (41ms)
[percy:core:discovery] - mimetype: image/svg+xml (0ms)
[percy:core:page] Page navigated (1ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Resize page to 1280x1024 @1x (100ms)
[percy:core:discovery] Wait for 100ms idle (2ms)
[percy:core:page] Page closed (105ms)
[percy:core:snapshot] --------- (2988ms)
[percy:core:snapshot] Received snapshot: Build Details/FileTree/Tree: With Multiple Levels (1ms)
[percy:core:snapshot] - url: http://localhost:59641/iframe.html?id=build-details-filetree-tree--with-multiple-levels (0ms)
[percy:core:snapshot] - widths: 375px, 1280px (0ms)
[percy:core:snapshot] - minHeight: 1024px (0ms)
[percy:core:snapshot] - discovery.allowedHostnames: localhost (0ms)
[percy:core:snapshot] - domSnapshot: true (0ms)
[percy:core] Discovering resources: Build Details/FileTree/Tree: With Multiple Levels (5ms)
[percy:core:page] Page created (5ms)
[percy:core:page] Taking snapshot: Build Details/FileTree/Tree: With Size Changes (76ms)
[percy:core:discovery] Wait for 100ms idle (1ms)
[percy:core:page] Resize page to 375x1024 @1x (25ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html?id=build-details-filetree-tree--with-multiple-levels (7ms)
[percy:core:discovery] Handling request: http://localhost:59641/iframe.html?id=build-details-filetree-tree--with-multiple-levels (3ms)
[percy:core:discovery] - Serving root resource (0ms)
[percy:core:discovery] Handling request: http://localhost:59641/static/media/folder.ac056cd0.svg (38ms)
[percy:core:discovery] - Resource cache hit (1ms)
[percy:core:discovery] Handling request: http://localhost:59641/static/media/file.a0c51867.svg (2ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:page] Serialize DOM (26ms)
[percy:core:discovery] Handling request: http://localhost:59641/fonts/circularstd-book.woff2 (25ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:page] Page navigated (33ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Resize page to 1280x1024 @1x (100ms)
[percy:core:discovery] Wait for 100ms idle (3ms)
[percy:core:page] Page closed (104ms)
[percy:core:snapshot] --------- (12497ms)
[percy:core:snapshot] Received snapshot: Build Details/FileTree/Tree: With Size Changes (0ms)
[percy:core:snapshot] - url: http://localhost:59641/iframe.html?id=build-details-filetree-tree--with-size-changes (0ms)
[percy:core:snapshot] - widths: 375px, 1280px (0ms)
[percy:core:snapshot] - minHeight: 1024px (0ms)
[percy:core:snapshot] - discovery.allowedHostnames: localhost (0ms)
[percy:core:snapshot] - domSnapshot: true (0ms)
[percy:core] Discovering resources: Build Details/FileTree/Tree: With Size Changes (8ms)
[percy:core:page] Page created (3ms)
[percy:core:page] Resize page to 375x1024 @1x (101ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html?id=build-details-filetree-tree--with-size-changes (7ms)
[percy:core:discovery] Handling request: http://localhost:59641/iframe.html?id=build-details-filetree-tree--with-size-changes (3ms)
[percy:core:discovery] - Serving root resource (0ms)
[percy:core:page] Taking snapshot: Build Details/FileTree/Folder: Primary (61ms)
[percy:core:discovery] Wait for 100ms idle (1ms)
[percy:core:discovery] Handling request: http://localhost:59641/static/media/folder.ac056cd0.svg (2ms)
[percy:core:discovery] - Resource cache hit (1ms)
[percy:core:discovery] Handling request: http://localhost:59641/static/media/file.a0c51867.svg (2ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:discovery] Handling request: http://localhost:59641/fonts/circularstd-book.woff2 (69ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:page] Serialize DOM (28ms)
[percy:core:page] Page navigated (13ms)
[percy:core:discovery] Wait for 100ms idle (1ms)
[percy:core:page] Resize page to 1280x1024 @1x (99ms)
[percy:core:discovery] Wait for 100ms idle (3ms)
[percy:core:page] Page closed (105ms)
[percy:core:snapshot] --------- (50627ms)
[percy:core:snapshot] Received snapshot: Build Details/FileTree/Folder: Primary (0ms)
[percy:core:snapshot] - url: http://localhost:59641/iframe.html?id=build-details-filetree-folder--primary (0ms)
[percy:core:snapshot] - widths: 375px, 1280px (0ms)
[percy:core:snapshot] - minHeight: 1024px (0ms)
[percy:core:snapshot] - discovery.allowedHostnames: localhost (0ms)
[percy:core:snapshot] - domSnapshot: true (0ms)
[percy:core] Discovering resources: Build Details/FileTree/Folder: Primary (13ms)
[percy:core:page] Page created (12ms)
[percy:core:page] Resize page to 375x1024 @1x (94ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html?id=build-details-filetree-folder--primary (6ms)
[percy:core:discovery] Handling request: http://localhost:59641/iframe.html?id=build-details-filetree-folder--primary (4ms)
[percy:core:discovery] - Serving root resource (0ms)
[percy:core:discovery] Handling request: http://localhost:59641/static/media/folder.ac056cd0.svg (97ms)
[percy:core:discovery] - Resource cache hit (1ms)
[percy:core:discovery] Handling request: http://localhost:59641/fonts/circularstd-book.woff2 (113ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:page] Page navigated (41ms)
[percy:core:discovery] Wait for 100ms idle (1ms)
[percy:core:page] Taking snapshot: Gravity Dashboard/Screens/Login: Primary (33ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Resize page to 1280x1024 @1x (66ms)
[percy:core:discovery] Wait for 100ms idle (3ms)
[percy:core:page] Serialize DOM (34ms)
[percy:core:page] Page closed (70ms)
[percy:core:snapshot] --------- (202790ms)
[percy:core:snapshot] Received snapshot: Gravity Dashboard/Screens/Login: Primary (0ms)
[percy:core:snapshot] - url: http://localhost:59641/iframe.html?id=gravity-dashboard-screens-login--primary (0ms)
[percy:core:snapshot] - widths: 375px, 1280px (0ms)
[percy:core:snapshot] - minHeight: 1024px (0ms)
[percy:core:snapshot] - discovery.allowedHostnames: localhost (0ms)
[percy:core:snapshot] - domSnapshot: true (0ms)
[percy:core] Discovering resources: Gravity Dashboard/Screens/Login: Primary (24ms)
[percy:core:page] Page created (13ms)
[percy:core:page] Resize page to 375x1024 @1x (98ms)
[percy:core:page] Navigate to: http://localhost:59641/iframe.html?id=gravity-dashboard-screens-login--primary (6ms)
[percy:core:discovery] Handling request: http://localhost:59641/iframe.html?id=gravity-dashboard-screens-login--primary (3ms)
[percy:core:discovery] - Serving root resource (0ms)
[percy:core:discovery] Handling request: http://localhost:59641/icons/github.svg (183ms)
[percy:core:discovery] - Resource cache hit (1ms)
[percy:core:discovery] Handling request: http://localhost:59641/icons/gravity_white_logo.svg (1ms)
[percy:core:discovery] Handling request: http://localhost:59641/static/media/login_blue_background.0ca69702.jpg (209ms)
[percy:core:discovery] Handling request: http://localhost:59641/fonts/circularstd-book.woff2 (0ms)
[percy:core:discovery] - Resource cache hit (0ms)
[percy:core:discovery] Processing resource: http://localhost:59641/icons/gravity_white_logo.svg (4ms)
[percy:core:discovery] - sha: 3a0fad4947b34b3928663ffcb9e5fe78413f71bf7ed01c4b46fe0eec89680681 (31ms)
[percy:core:discovery] - mimetype: image/svg+xml (0ms)
[percy:core:discovery] Processing resource: http://localhost:59641/static/media/login_blue_background.0ca69702.jpg (1ms)
[percy:core:page] Page navigated (6ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:discovery] - sha: 95fe82d1c6c61e7e4ad46dec3f66b83af63dd1b94896e13b861276ed5dcef059 (6ms)
[percy:core:discovery] - mimetype: image/jpeg (1ms)
[percy:core:page] Resize page to 1280x1024 @1x (105ms)
[percy:core:page] Taking snapshot: Build Details/Tag: With Link And Icon Hover (3ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:discovery] Wait for 100ms idle (0ms)
[percy:core:page] Page closed (106ms)
</details>

## Code to reproduce issue

We're running `percy storybook ./storybook-static` against this Storybook build:

[storybook-static.zip](https://github.com/percy/percy-storybook/files/10458702/storybook-static.zip)

**Comment by itsjwala** [comment]
Hey @marcoow 👋 

* can you try lowering the concurrency? it seems like default [concurrency](https://docs.percy.io/docs/cli-configuration#a-complete-config) of `10` might be causing CPU contention. 

```
discovery:
  concurrency: 2
```

**Reply by itsjwala** [reply]
secondly, you can try enabling javascript, this won't take dom snapshots and upload the stories source directly.

```
snapshot:
  enable-javascript: true
```

**Comment by samarsault** [comment, marked as answer]
@marcoow I see some infinite recursion in storybook rendering process; its coming from a `rewriteStylesheet` function

![image](https://user-images.githubusercontent.com/2987087/221120026-069ffb9e-ddb3-4d38-a5e4-e494847825c4.png)

On searching Github in context of storybook, I see this is part of the pseudo states addon.
Are you using [pseudo state addon](https://github.com/chromaui/storybook-addon-pseudo-states)? If yes can you try running a build after removing it?

**Reply by marcoow** [reply]
It seems to be this indeed – disable pseudo states fixes it…
